tests/test_types.py

```diff
@@ -231,6 +246,27 @@ def test_nested_field() -> None:
     assert "validation errors for NestedField" in str(exc_info.value)
 
 
+def test_nested_field_type_as_str_unsupported() -> None:
+    unsupported_types = ["list", "map", "struct"]
+    for type_str in unsupported_types:
+        with pytest.raises(ValueError) as exc_info:
+            _ = NestedField(1, "field", type_str, required=True)
+        assert f"Unsupported field type: '{type_str}'" in str(exc_info.value)
+
+
+@pytest.mark.parametrize("type_str, type_class", primitive_types.items())
+def test_nested_field_type_as_str(type_str: str, type_class: type) -> None:
```

Comment:
nit:
```suggestion
def test_nested_field_primitive_type_as_str(type_str: str, type_class: type) -> None:
```

Reply:
nit: parametrizing the test actually is slower compared to just running a for loop into the function
```

for type_str, type_class in primitive_types.items():
    field_var = NestedField(
        1,
        "field",
        type_str,
        required=True,
    )
    assert isinstance(
        field_var.field_type, type_class
    ), f"Expected {type_class.__name__}, got {field_var.field_type.__class__.__name__}"
```